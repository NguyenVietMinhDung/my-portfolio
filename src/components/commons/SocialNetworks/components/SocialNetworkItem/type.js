export type Props = {
  name: string,
  url: string,
  iconUrl: string,
};
