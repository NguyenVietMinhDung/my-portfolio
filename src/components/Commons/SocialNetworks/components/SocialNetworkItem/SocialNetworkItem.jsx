// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './SocialNetworkItem.scss';
import type { Props } from './type';

const SocialNetworkItem = (props: Props) => {
  const { url, iconUrl } = props;
  return (
    <li>
      <a href={url}>
        <img styleName="social-network-item" src={iconUrl} alt="" />
      </a>
    </li>
  );
};

export default memo<Props>(CSSModules(SocialNetworkItem, styles));
