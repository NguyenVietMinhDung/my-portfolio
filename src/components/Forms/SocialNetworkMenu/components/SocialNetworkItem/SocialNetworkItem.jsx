// @flow
import React from 'react';
import CSSModules from 'react-css-modules';
import styles from './SocialNetworkItem.scss';
import type { SocialNetworkItemProps } from './type';

const SocialNetworkItem = (props: SocialNetworkItemProps) => {
  const { url, iconUrl } = props;
  return (
    <li>
      <a href={url}>
        <img styleName="social-network-item" src={iconUrl} alt="" />
      </a>
    </li>
  );
};

export default CSSModules(SocialNetworkItem, styles);
