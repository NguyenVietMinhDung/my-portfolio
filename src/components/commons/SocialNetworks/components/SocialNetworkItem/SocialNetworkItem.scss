.social-network-item {
	width: 40px;
	height: 40px;
	padding: 0 5px;
}
