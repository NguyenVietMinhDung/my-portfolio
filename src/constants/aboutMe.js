export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
