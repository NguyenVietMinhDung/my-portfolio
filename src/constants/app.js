import facebookIconUrl from '../assets/images/facebook.png';
import instagramIconUrl from '../assets/images/instagram.png';
import linkedInIconUrl from '../assets/images/linkedin.png';

export const LOADER_SIZE = 50;
export const LOADER_SIZE_UNIT = 'px';
export const LOADER_COLOR = '#754ef9';
export const socialNetworks = [
  {
    name: 'Facebook',
    url: 'https://www.facebook.com/minh.dung.391',
    iconUrl: facebookIconUrl,
  },
  {
    name: 'Instagram',
    url: 'https://www.instagram.com/dungvmnguyen',
    iconUrl: instagramIconUrl,
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/dungvmn1995',
    iconUrl: linkedInIconUrl,
  },
];
export logoUrl from '../assets/images/logo.png';
