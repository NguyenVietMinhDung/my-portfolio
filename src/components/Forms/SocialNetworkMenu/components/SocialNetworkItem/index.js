import SocialNetworkItem from './SocialNetworkItem';

export default SocialNetworkItem;
