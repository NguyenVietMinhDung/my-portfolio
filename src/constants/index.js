import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
