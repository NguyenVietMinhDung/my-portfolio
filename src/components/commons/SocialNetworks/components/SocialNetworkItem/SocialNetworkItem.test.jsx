import React from 'react';
import { shallow } from 'enzyme';
import { socialNetworks } from '../../../../../constants/app';
import SocialNetworkItem from './SocialNetworkItem';

const defaultProps = {
  url: socialNetworks[0].url,
  iconUrl: socialNetworks[0].iconUrl,
};

describe('<SocialNetworkItem />', () => {
  test('renders correctly', () => {
    const component = shallow(<SocialNetworkItem {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
