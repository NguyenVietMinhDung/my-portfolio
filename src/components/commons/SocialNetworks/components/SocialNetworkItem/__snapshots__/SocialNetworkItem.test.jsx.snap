// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<SocialNetworkItem /> renders correctly 1'] = `
<li>
  <a
    href="https://www.facebook.com/minh.dung.391"
  >
    <img
      alt=""
      className="social-network-item"
      src="test-file-stub"
    />
  </a>
</li>
`;
